crates/re_viewer_context/src/time_control.rs

```diff
@@ -477,47 +487,49 @@ impl TimeControl {
 
     /// Is the current time in the selection range (if any), or at the current time mark?
     pub fn is_time_selected(&self, timeline: &Timeline, needle: TimeInt) -> bool {
-        if timeline != &self.timeline {
+        if timeline != self.timeline.get() {
             return false;
         }
 
-        if let Some(state) = self.states.get(&self.timeline) {
+        if let Some(state) = self.states.get(self.timeline.get()) {
             state.time.floor() == needle
         } else {
             false
         }
     }
 
     pub fn set_timeline_and_time(&mut self, timeline: Timeline, time: impl Into<TimeReal>) {
-        self.timeline = timeline;
+        self.timeline = EditableAutoValue::UserEdited(timeline);
         self.set_time(time);
     }
 
     pub fn set_time(&mut self, time: impl Into<TimeReal>) {
         let time = time.into();
 
         self.states
-            .entry(self.timeline)
+            .entry(*self.timeline.get())
             .or_insert_with(|| TimeState::new(time))
             .time = time;
     }
 
     /// The range of time we are currently zoomed in on.
     pub fn time_view(&self) -> Option<TimeView> {
-        self.states.get(&self.timeline).and_then(|state| state.view)
+        self.states
+            .get(self.timeline.get())
+            .and_then(|state| state.view)
     }
 
     /// The range of time we are currently zoomed in on.
     pub fn set_time_view(&mut self, view: TimeView) {
         self.states
-            .entry(self.timeline)
+            .entry(*self.timeline.get())
             .or_insert_with(|| TimeState::new(view.min))
             .view = Some(view);
     }
 
     /// The range of time we are currently zoomed in on.
     pub fn reset_time_view(&mut self) {
-        if let Some(state) = self.states.get_mut(&self.timeline) {
+        if let Some(state) = self.states.get_mut(self.timeline.get()) {
```

Comment:
Instead of changing all `&foo` to `foo.get()`, just implement `Deref` for `EditableAutoValue`!

Reply:
for those cases I still need `get()` due to the way Deref works in rust: `*self.timeline` gives the underlying `Timeline` by value. But with `Deref` in place `*self.timeline.get()` becomes `*self.timeline` and `self.timeline.get().typ()` becomes `self.timeline.typ()`

Reply:
could ofc also do `&*self.timeline.get()` if that's preferred

Reply:
I mean do `&*self.timeline`